pkg/cmd/workspace/create.go

```diff
@@ -199,11 +233,44 @@ func init() {
 	CreateCmd.Flags().StringVar(&providerFlag, "provider", "", "Specify the provider (e.g. 'docker-provider')")
 	CreateCmd.Flags().StringVarP(&ideFlag, "ide", "i", "", "Specify the IDE ('vscode' or 'browser')")
 	CreateCmd.Flags().StringVarP(&targetNameFlag, "target", "t", "", "Specify the target (e.g. 'local')")
+	CreateCmd.Flags().StringVar(&customImageFlag, "custom-image", "", "Create the project with the custom image passed as the flag value; Requires setting --custom-image-user flag as well")
+	CreateCmd.Flags().StringVar(&customImageUserFlag, "custom-image-user", "", "Create the project with the custom image user passed as the flag value; Requires setting --custom-image flag as well")
+	CreateCmd.Flags().StringVar(&devcontainerPathFlag, "devcontainer-path", "", "Automatically assign the devcontainer builder with the path passed as the flag value")
+	CreateCmd.Flags().StringVar(&builderFlag, "builder", "", fmt.Sprintf("Specify the builder (currently %s/%s/%s)", string(create.AUTOMATIC), string(create.DEVCONTAINER), string(create.NONE)))
+
 	CreateCmd.Flags().BoolVar(&manualFlag, "manual", false, "Manually enter the git repositories")
 	CreateCmd.Flags().BoolVar(&multiProjectFlag, "multi-project", false, "Workspace with multiple projects/repos")
 	CreateCmd.Flags().BoolVarP(&codeFlag, "code", "c", false, "Open the workspace in the IDE after workspace creation")
 }
 
+func validateFlags() error {
```

Comment:
Please check if it is feasible to use built-in cobra methods cmd.`MarkFlagsMutuallyExclusive` and cmd.`MarkFlagsRequiredTogether` for some of these checks.